hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/payload/ExpressionPayload.scala

```diff
@@ -270,6 +270,17 @@ class ExpressionPayload(@transient record: GenericRecord,
     isDeletedRecord || isDeleteOnCondition
   }
 
+  override def getIndexedRecord(schema: Schema, properties: Properties): HOption[IndexedRecord] = {
+    val recordSchema = getRecordSchema(properties)
```

Comment:
can we use the passed in schema instead?

Reply:
The schema in the properties is what matches the subset of fields specified in the payload.